packages/timeline-state-resolver/src/integrations/tricaster/triCasterStateDiffer.ts

```diff
@@ -300,21 +301,29 @@ export class TriCasterStateDiffer {
 		source: CommandName.CROSSPOINT_SOURCE,
 	}
 
-	private effectCommandGenerator: CommandGeneratorFunction<
+	private keyerEffectCommandGenerator: CommandGeneratorFunction<
 		TriCasterTransitionEffect,
-		RequiredDeep<TriCasterMixEffectState | TriCasterKeyerState>
-	> = ({ value, target, state }) => {
-		const commands: TriCasterCommand[] = []
-		if (value === 'cut') {
-			return commands
-		}
-		if (typeof value === 'number') {
-			commands.push({ name: CommandName.SELECT_INDEX, target, value })
-		} else if (value === 'fade') {
-			commands.push({ name: CommandName.SELECT_INDEX, target, value: 1 })
+		RequiredDeep<TriCasterKeyerState>
+	> = this.effectCommandGenerator(CommandName.SELECT_INDEX)
+
+	private mixEffectEffectCommandGenerator: CommandGeneratorFunction<
+		TriCasterTransitionEffect,
+		RequiredDeep<TriCasterMixEffectState>
+	> = this.effectCommandGenerator(CommandName.SET_MIX_EFFECT_BIN_INDEX)
+
+	private effectCommandGenerator(
+		selectCommand: TriCasterGenericCommandName<number>
+	): CommandGeneratorFunction<TriCasterTransitionEffect, RequiredDeep<TriCasterKeyerState | TriCasterMixEffectState>> {
+		return ({ value, target, state }) => {
```

Comment:
Types are sorely needed here

Reply:
Not all inferred types are bad. These are dictated by `CommandGeneratorFunction` and I feel like it would be redundant to put them on the returned function too. This will be refactored slightly in https://github.com/tv2/tv-automation-state-timeline-resolver/pull/50 to make the type slightly more readable